### PR TITLE
Updating story for the Button component.

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes  [DSD-88](https://jira.nypl.org/browse/DSD-88)

## **This PR does the following:**
- Updates the story for the `Button` component to display patterns shown in the respective figma file.
- This also adds the update from the ticket description which is adding Controls to change the width of the text inside a button, specifically for the "Button Group".

### Front End Review:
- [ ] View [the basic example in Storybook](https://deploy-preview-517--stoic-murdock-c7f044.netlify.app/?path=/story/button--basic)
- [ ] View [all docs](https://deploy-preview-517--stoic-murdock-c7f044.netlify.app/?path=/docs/button--basic)

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Do documentation-only PRs not trigger the netlify deployment action?

**Comment by hokei** [comment]
Just checking - it eventually deployed right?  Or did you have to do something specific with it?

**Comment by EdwinGuzman** [comment]
It eventually deployed. I'm not sure if this is related but I noticed that _this_ built when I created another PR.

File: src/components/Button/Button.stories.mdx [hunk 1]
```diff
@@ -31,55 +31,203 @@ import { action } from "@storybook/addon-actions";
...
+## Patterns
+
+Buttons are blue by default but should be red when it is intended as a call out.
+The "Back to Top" button should have secondary button styles and an up arrow icon.
+
+<Canvas>
+  <Story name="Patterns">
+    <Button
+      buttonType={ButtonTypes.Primary}
+      attributes={{ style: { backgroundColor: "#d0343a" } }}
```
Comment (kaplanmichael) [inline src/components/Button/Button.stories.mdx]: Please use color variable.

File: src/components/Button/Button.stories.mdx [hunk 2]
```diff
@@ -31,55 +31,203 @@ import { action } from "@storybook/addon-actions";
...
+    >
+      <Icon name={IconNames.close} decorative={true} modifiers={["small"]} />
+    </Button>
+  </div>
+</Canvas>
+
+## Patterns
+
+Buttons are blue by default but should be red when it is intended as a call out.
+The "Back to Top" button should have secondary button styles and an up arrow icon.
```
Comment (kaplanmichael) [inline src/components/Button/Button.stories.mdx]: The "Back to Top" button must be of type "Secondary" and must include an up arrow icon.

**Review by kaplanmichael — CHANGES_REQUESTED** [review]
Looks good.  Just a few nits.

File: src/docs/Intro.stories.mdx [hunk 3]
```diff
@@ -30,7 +31,7 @@ one example of how we use the NYPL Design System on NYPL.org.
       alt={""}
     />
   }
-  backgroundColor="#d0343a"
+  backgroundColor={getCSSVariable("--section-books-and-more-primary")}
   foregroundColor="#ffffff"
```
Comment (kaplanmichael) [inline src/docs/Intro.stories.mdx]: Would you mind also adding  "--ui-white" instead of #ffffff ?

**Review by kaplanmichael — CHANGES_REQUESTED** [review]
One quick one and then good to go!

**Review by kaplanmichael — APPROVED** [review]

**Pull request merged**